lib/libc/common/source/stdlib/malloc.c

```diff
@@ -251,6 +275,15 @@ void *realloc(void *ptr, size_t size)
 	ARG_UNUSED(ptr);
 	return malloc(size);
 }
+
+struct mallinfo2 mallinfo2(void)
```

Comment:
There is no need to add this stub for mallinfo if malloc heap size is zero.

The main reason the malloc stub exists is because malloc is part of ISO C and apps still need to be able to link